static/openapi/eds-spec.json

```diff
@@ -0,0 +1,399 @@
+          },
+          {
+            "in": "query",
+            "name": "type",
+            "required": true,
+            "schema": {
+              "type": "string"
+            },
+            "description": "The type of the stream (local or global or collection)."
+          }
+        ],
+        "requestBody": {
+          "required": true,
+          "description": "The request body containing the event data to publish to the stream.",
+          "content": {
+            "application/json": {
+              "schema": {}
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successfully published events.",
+            "headers": {
+              "x-gdn-service": {
+                "description": "The hostname of the GDN.",
+                "schema": {
+                  "type": "string",
+                  "description": "Custom header value indicating the hostname of the GDN."
+                }
+              },
+              "X-PhotonIQ-ES": {
+                "description": "Epoch time header.",
+                "schema": {
+                  "type": "string",
+                  "description": "Epoch time in seconds."
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "properties": {
+                    "error": {
+                      "type": "string",
+                      "description": "Error message."
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "ws://api/es/v1/subscribe": {
+      "get": {
+        "summary": "Subscribe to stream",
+        "description": "Subscribe to stream via WebSocket based on stream, type, or query parameters. Provide either `stream` or `filters` but not both or neither.\nThis API keeps the connection live and continuously receives messages from the server based on the specified `stream` or `filters`.\n\"filters\\\" is a JSON format string with different fields. E.g :- <br>\nfilters={ <br>\n  \\\"action\\\":“add”, <br>\n  \\\"once\\\":”TRUE | FALSE”, <br>\n  \\\"queries\\\":[ <br>\n  “select * from XXXX where n1==v1”, <br>\n  “select * from MMMM where n2==v1 OR n3>5”, <br>\n  “select \\<path\\> from XXXX” <br>\n  ] <br>\n  } <br>\n  The *path* is a dot separated fully qualified string representing a json path for an object. *once* flag (default is false) specifies if the filters needs to be applied just once. <br>\n  *filters* can be dynamically added and removed by sending message data on web socket connection in JSON format. E.g :- <br>\n  {\\\"action\\\": \\\"remove\\\", \\\"queries\\\": [ \\\"select * from XXXX where n1=v1\\\"]}\n\n  **This API should not be tried from the Swagger UI**.\n",
```

Comment:
```suggestion
        "description": "Subscribe to stream via WebSocket based on stream, type, or query parameters. Provide either `stream` or `filters`.\nThis API keeps the connection live and continuously receives messages from the server based on the specified `stream` or `filters`.\n\"filters\\\" is a JSON format string with different fields. E.g :- <br>\nfilters={ <br>\n  \\\"action\\\":“add”, <br>\n  \\\"once\\\":”TRUE | FALSE”, <br>\n  \\\"queries\\\":[ <br>\n  “select * from XXXX where n1==v1”, <br>\n  “select * from MMMM where n2==v1 OR n3>5”, <br>\n  “select \\<path\\> from XXXX” <br>\n  ] <br>\n  } <br>\n  The *path* is a dot separated fully qualified string representing a JSON path for an object. *once* flag (default is false) specifies if the filters needs to be applied just once. <br>\n  *filters* can be dynamically added and removed by sending message data on web socket connection in JSON format. E.g :- <br>\n  {\\\"action\\\": \\\"remove\\\", \\\"queries\\\": [ \\\"select * from XXXX where n1=v1\\\"]}\n\n  **This API should not be tried from the Swagger UI**.\n",
```

Reply:
Maybe something about how you need to use this with a WebSocket-compatible method? Not sure what the right wording is, but Manoj mentioned that curl, for example, wouldn't work because it doesn't work with WebSocket.

Reply:
As for "e.g.," I hate it and generally replace it with "such as."

Reply:
Yes, the entire JSON is passed as query params as that is what was asked in the PRD. And true, this cannot be tried from UI as WS is not supported. Also replaced e.g with - For example.

Reply:
I have no clue on the URL thing as the eds spec json as the proper host entry. Could this be due to WS not be supported ?